src/RadhydroSimulation.hpp

```diff
@@ -1100,9 +1133,22 @@ auto RadhydroSimulation<problem_t>::advanceHydroAtLevel(amrex::MultiFab &state_o
 		amrex::iMultiFab redoFlag(grids[lev], dmap[lev], 1, 1);
 		redoFlag.setVal(quokka::redoFlag::none);
 
+		if constexpr (Physics_Traits<problem_t>::is_mhd_enabled) {
+			std::array<std::array<amrex::MultiFab, 2>, AMREX_SPACEDIM> emf_comps_ec;
+			for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+				emf_comps_ec[idim][0].define(
+				    amrex::convert(ba_cc, amrex::IntVect::TheDimensionVector(idim) + amrex::IntVect::TheDimensionVector((idim + 1) % 3)), dm, 1,
+				    nghost_fc_);
+				emf_comps_ec[idim][1].define(
+				    amrex::convert(ba_cc, amrex::IntVect::TheDimensionVector(idim) + amrex::IntVect::TheDimensionVector((idim + 2) % 3)), dm, 1,
+				    nghost_fc_);
+				fspds[idim].FillBoundary(geom[lev].periodicity());
```

Comment:
Just a note for the future: this will have to be modified for AMR to use `FillPatchTwoLevels` instead (since the wavespeeds may be at a coarse-fine boundary).